src/lib/RuleResultRecombinator.ts

```diff
@@ -111,11 +123,15 @@ ${v.message.trim()}
 			column,
 			url
 		} = violation;
-		return `<testcase name="${fileName}">
-<failure message="${fileName}: ${line} ${htmlEscaper.escape(message.trim())}" type="${severity}">
-${severity}: ${message.trim()}
-Category: ${category} - ${ruleName}
-File: ${fileName}
+		const escapedFileName = this.safeHtmlEscape(fileName);
```

Comment:
Being overly aggressive, because filenames can have characters that need to be escaped.